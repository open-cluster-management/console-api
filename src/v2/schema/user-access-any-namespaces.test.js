/** *****************************************************************************
 * Copyright (c) 2020 Red Hat, Inc.
 ****************************************************************************** */

import supertest from 'supertest';
import server, { GRAPHQL_PATH } from '../index';

describe('User Access All Namespaces Resolver', () => {
  test('Correctly Resolves User Access All Namespaces Query', () => new Promise((done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
          {
            userAccessAnyNamespaces(resource:"pods", action:"delete", apiGroup:"")
          }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  }));
});
