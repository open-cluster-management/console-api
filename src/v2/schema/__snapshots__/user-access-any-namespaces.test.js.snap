// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Access All Namespaces Resolver Correctly Resolves User Access All Namespaces Query 1`] = `
Object {
  "data": Object {
    "userAccessAnyNamespaces": true,
  },
}
`;
